politeiawww/backend.go

```diff
@@ -78,10 +78,20 @@ type BackendProposalMetadata struct {
 	Version   uint64 `json:"version"`   // BackendProposalMetadata version
 	Timestamp int64  `json:"timestamp"` // Last update of proposal
 	Name      string `json:"name"`      // Generated proposal name
+	UserId    string `json:"userid"`    // ID of user who submitted proposal
 	PublicKey string `json:"publickey"` // Key used for signature.
```

Comment:
We can calculate userid from the public key.  My worry is that this will get out of whack and may need reconciliation later. Any objections fo figuring out the user id from pubkey?

Reply:
Nope, no objections - it will be slower to derive the userid from the publickey because we have to scan all public keys of all users for every proposal, but that shouldn't be a problem because it only needs to happen when fetching inventory.

Reply:
I am really torn about this. From experience reconciling, one or more identifiers always ends in tears. Any other suggestions?

Reply:
Discussed on Slack - we'll build an in-memory pubkey-userid map on startup and use it for lookups.